dandi/upload.py

```diff
@@ -61,6 +61,21 @@ def upload(
             " paths.  Use 'dandi download' or 'organize' first."
         )
 
+    # pre-validate BIDS datasets before going for individual
+    # files etc
+    bids_datasets = _bids_discover_and_validate(dandiset_.path, paths, validation)
+
+    if bids_datasets:
+        bids_datasets = [str(i) for i in bids_datasets]
```

Comment:
Aren't the elements of `bids_datasets` already strings?

Reply:
No, they are `PosixPath` objects.